crates/bevy_render/src/render_resource/pipeline_cache.rs

```diff
@@ -386,3 +406,49 @@ impl RenderPipelineCache {
         }
     }
 }
+
+fn log_shader_error(source: &ProcessedShader, err: &AsModuleDescriptorError) {
+    use codespan_reporting::{
+        diagnostic::{Diagnostic, Label},
+        files::SimpleFile,
+        term,
+    };
+
+    match source {
+        ProcessedShader::Wgsl(source) => match err {
```

Comment:
clippy says this single match should be an `if let`. I personally disagree and find this match more readable.

Reply:
Hmm yeah i dont have strong opinions here, but deferring to clippy gives us consistency. Happy to revisit this rule in a separate issue if you want to make your case :)